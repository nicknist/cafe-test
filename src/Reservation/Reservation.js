import React from 'react';

const Reservation = ({ reservation, deleteReservation }) => {
  if(!reservation) {
    return <h1>Loading...</h1>
  }

  return (
    <div className='single-res'>
      <h2>{reservation.name}</h2>
      <p>{reservation.date}</p>
      <p>{reservation.time}pm</p>
      <p>Number of Guests: {reservation.number}</p>
      <button
        onClick={() => deleteReservation(reservation)} 
        className='cancel'
      >Cancel</button>
    </div>
  )
}

export default Reservation;
