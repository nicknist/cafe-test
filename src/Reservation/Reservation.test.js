import React from 'react';
import ReactDOM from 'react-dom';
import Reservation from './Reservation';
import { shallow } from 'enzyme';

describe('Reservation', () => {
  let wrapper;
  let mockDeleteReservation;
  let mockReservation;

  beforeEach(() => {
    mockReservation = {name: '1', time: '2'};
    mockDeleteReservation = jest.fn();
    wrapper = shallow(<Reservation reservation={mockReservation} deleteReservation={mockDeleteReservation}/>);
  })

  it('should match the snapshot when loading', () => {
    let unLoaded = shallow(<Reservation />)
    expect(unLoaded).toMatchSnapshot();
  })

  it('should match snapshot when it has reservations', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should run deleteReservation when cancel button is clicked', () => {
    wrapper.find('button').simulate('click');

    expect(mockDeleteReservation).toHaveBeenCalledWith(mockReservation)
  })
})
