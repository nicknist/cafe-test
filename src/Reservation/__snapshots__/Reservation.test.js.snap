// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Reservation should match snapshot when it has reservations 1`] = `
<div
  className="single-res"
>
  <h2>
    1
  </h2>
  <p />
  <p>
    2
    pm
  </p>
  <p>
    Number of Guests: 
  </p>
  <button
    className="cancel"
    onClick={[Function]}
  >
    Cancel
  </button>
</div>
`;

exports[`Reservation should match the snapshot when loading 1`] = `
<h1>
  Loading...
</h1>
`;
